### PR TITLE
finality: sudo message for `BeginBlock`

### LINKED ISSUE #2: BeginBlock for babylon-contract
- Babylon SDK triggers Babylon contracts upon BeginBlock/EndBlock (sudo call)

### PR DESCRIPTION
**Pull request by SebastianElvis**
Closes #2 

This PR provides sudo message `BeginBlock` for Babylon contracts. The sudo message will be sent to BTC staking contract upon each `BeginBlock`.

To test, run `make test-e2e`. Note that the wasm files are generated on https://github.com/babylonchain/babylon-contract/pull/123

### CODE REVIEW

File: tests/testdata/version.txt
Comment (maurolacy) [inline tests/testdata/version.txt:1]: Perhaps this can now be replaced by the release tag.

File: x/babylon/keeper/msg_server.go [hunk 1]
```diff
@@ -14,3 +19,20 @@ type msgServer struct {
 func NewMsgServer(k *Keeper) *msgServer {
 	return &msgServer{k: k}
 }
+
+// UpdateParams updates the params.
+func (ms msgServer) UpdateParams(goCtx context.Context, req *types.MsgUpdateParams) (*types.MsgUpdateParamsResponse, error) {
```
Comment (maurolacy) [inline x/babylon/keeper/msg_server.go:24]: Nit: Shouldn't this belong to a different / independent PR?

Or this is needed to get the contracts addresses?

File: x/babylon/contract/out_message.go [hunk 2]
```diff
@@ -1,7 +1,16 @@
...
+
 // SudoMsg is a message sent from the Babylon module to a smart contract
-// TODO: implement
 type SudoMsg struct {
-	TestSudoMsg *struct{} `json:"test_sudo_msg,omitempty"`
+	BeginBlockMsg *BeginBlock `json:"begin_block,omitempty"`
+}
+
+type BeginBlock struct {
+	Height     int64     `json:"height"`       // Height returns the height of the block
```
Comment (maurolacy) [inline x/babylon/contract/out_message.go]:
```suggestion
	Height     int64     `json:"height"`       // Height is the height of the block
```
?

(The same with the other fields)

**Review by maurolacy — APPROVED** [review]
LGTM.

File: x/babylon/keeper/msg_server.go (hunk 1 above)
Comment (SebastianElvis) [reply x/babylon/keeper/msg_server.go:24]: It is necessary as eventually one might need to trigger gov prop to call this message in order to set up the Babylon contract addresses. It's not a big change -- most things about this message handler are copied from Babylon e.g., https://github.com/babylonchain/babylon/blob/2a7c91d6f1649912f14667d3a3ee6a2c1792c3d9/x/btccheckpoint/keeper/msg_server.go#L97-L114

**Comment by SebastianElvis** [comment]
CI failed due to config reasons. Will fix in separate PRs

**Pull request merged**